src/%IPM/Main.cls

```diff
@@ -1693,11 +1693,28 @@ ClassMethod Load(ByRef pCommandInfo) [ Internal ]
 		set tDirectoryName=..LoadFromRepo(tDirectoryName,.tParams)
 	}
 	If ##class(%File).DirectoryExists(tDirectoryName) {
-		$$$ThrowOnError(##class(%IPM.Utils.Module).LoadNewModule(tDirectoryName,.tParams))
+		Set tSC = $$$OK
+		Try {
+			Lock +^IPM.ModuleInstallLock(1):2
+			Set tSC = ##class(%IPM.Utils.Module).LoadNewModule(tDirectoryName,.tParams)
+		} Catch e {
+			Set tSC=$$$ADDSC(tSC,e.AsStatus())
+		}
+		Lock -^IPM.ModuleInstallLock(1)
+		$$$ThrowOnError(tSC)
 	} ElseIf ##class(%File).Exists(tDirectoryName),$$$lcase($PIECE(tDirectoryName, ".", *)) = "tgz" {
 		Set tTargetDirectory = $$$FileTempDirSys
 		Set tSC = ##class(%IPM.General.Archive).Extract(tDirectoryName, tTargetDirectory)
-		$$$ThrowOnError(##class(%IPM.Utils.Module).LoadNewModule(tTargetDirectory, .tParams))
+		$$$ThrowOnError(tSC)
+		Set tSC = $$$OK
+		Try {
+			Lock +^IPM.ModuleInstallLock(1):2
```

Comment:
If put in class %IPM.Utils.Module's methods, then we would need to lock all of the methods there, so not only LoadNewModule, but also LoadModuleReference, LoadDependencies, etc. since they don't really have a relationship that one method is only called in another, I feel like put the lock in the starting point (install, load in Main class) would be cleaner. But I'm open to discuss this